lib/streamlit/testing/v1/app_test.py

```diff
@@ -90,6 +90,64 @@
 
 
 class AppTest:
+    """
+    A simulated Streamlit app to check the correctness of displayed\
+    elements and outputs.
+
+    An instance of ``AppTest`` simulates a running Streamlit app. This class
+    provides methods to set up, manipulate, and inspect the app contents via
+    API instead of a browser UI. It can be used to write automated tests of an
+    app in various scenarios. These can then be run using a tool like pytest.
+
+    ``AppTest`` can be initialized by one of three class methods:
+
+    * |st.testing.v1.AppTest.from_file|_ (recommended)
+    * |st.testing.v1.AppTest.from_string|_
+    * |st.testing.v1.AppTest.from_function|_
+
+    Once initialized, Session State and widget values can be updated and the
+    script can be run. Unlike an actual live-running Streamlit app, you need to
+    call ``AppTest.run()`` explicitly to re-run the app after changing a widget
+    value.
+
+    ``AppTest`` enables developers to build tests on their app as-is, in the
+    familiar python test format, without major refactoring or abstracting out
+    logic to be tested separately from the UI. Tests can run quickly with very
+    low overhead. A typical pattern is to build a suite of tests for an app
+    that ensure consistent functionality as the app evolves, and run the tests
+    locally and/or in a CI environment like Github Actions.
+
+    .. note::
+        ``AppTest`` only supports testing a single page of an app per
+        instance. For multipage apps, each page will need to be tested
+        separately. No methods exist to programatically switch pages within
+        ``AppTest``.
+
+    .. |st.testing.v1.AppTest.from_file| replace:: ``st.testing.v1.AppTest.from_file``
+    .. _st.testing.v1.AppTest.from_file: #apptestfrom_file
+    .. |st.testing.v1.AppTest.from_string| replace:: ``st.testing.v1.AppTest.from_string``
+    .. _st.testing.v1.AppTest.from_string: #apptestfrom_string
+    .. |st.testing.v1.AppTest.from_function| replace:: ``st.testing.v1.AppTest.from_function``
+    .. _st.testing.v1.AppTest.from_function: #apptestfrom_function
+
+    Attributes
+    ----------
+    secrets: dict[str, Any]
+        Dictionary of secrets to be used the simulated app. Use dict-like
+        syntax to set secret values for the simulated app.
+
+    session_state: SafeSessionState
+        Session State for the simulated app. SafeSessionState object supports
+        read and write operations as usual for Streamlit apps.
+
+    query_params: dict[str, Any]
+        Dictionary of query parameters to be used by the simluated app. Use
+        dict-like syntax to set ``query_params`` values for the simulated app.
+
+        Note: Inspecting ``query_params`` after a simulated app run is not yet
+        supported.
```

Comment:
@AnOctopus should confirm, but I think this might be fixed now so it does actually work - see [here](https://github.com/streamlit/streamlit/blob/develop/lib/streamlit/testing/v1/app_test.py#L223-L225)

Reply:
I removed the note. I can always add it back in if I need to.